kolena/workflow/test_run.py

```diff
@@ -242,7 +245,10 @@ def upload_inferences(self, inferences: List[Tuple[TestSample, Inference]]) -> N
         df_serializable = df_validated.as_serializable()
 
         init_response = init_upload()
-        upload_data_frame_chunk(df_serializable, init_response.uuid)
+        batch_size = (
+            len(df_serializable) // MB_BATCH_LIMIT // (get_preflight_export_size(df_serializable) // MB_CONVERSION)
+        )
+        upload_data_frame(df_serializable, batch_size, init_response.uuid)
```

Comment:
could we create a wrapper func for this logic?

Reply:
yes doing that and adding tests